straxen/plugins/bayes_peak_classification.py

```diff
@@ -121,30 +130,63 @@ def compute_inference(bins: int, bayes_n_nodes: int, cpt: np.ndarray, n_bayes_cl
     waveform_num_bin_edges = len(waveform_bin_edges)
     quantile_bin_edges = bins[1, :][bins[1, :] > -1]
     quantile_num_bin_edges = len(quantile_bin_edges)
-
     waveform_values = np.digitize(waveforms, bins=waveform_bin_edges)-1
-    waveform_values[waveform_values < 0] = int(0)
-    waveform_values[waveform_values > int(waveform_num_bin_edges - 2)] = int(waveform_num_bin_edges - 2)
+    _set_2d_to_zero_or_max_val(waveform_values,
+                               min_val=0,
+                               max_val=np.int64(waveform_num_bin_edges - 2))
```

Comment:
Is not this the same as `np.clip` ?

Reply:
Okay, I am fine either way. To me the overall improvement is tiny so not sure if it justifies adding custom code. But the code is also rather simple so I also do not mind adding it.

Reply:
fair point